api/v1alpha1/secretstore_types.go

```diff
@@ -34,6 +34,14 @@ type SecretStoreSpec struct {
 	// +optional
 	Controller string `json:"controller"`
 
+	// Used to configure the provider. Only one provider may be set
+	Provider *SecretStoreProvider `json:"provider"`
```

Comment:
This part changes the `SecretStore` CRD spec from:
```yaml
...
spec:
  awssm:
    ...
```

To:
```yaml
...
spec:
  provider:
    awssm:
    ...
```

Are we okay with it? _I think_ it's fine, since it makes easier to use the `Min/MaxProperties` filter.

Do you think it has any impacts elsewhere? 

Any input @knelasevero ?

Reply:
If everything is fine, could u update the crd-spec too @moolen ?

Reply:
That's right i wanted to use the `Min/MaxProperties` filter, its just the `spec.controller` that mixes up with the provider configuration - so i rooted it under `provider`. sure, i'll update the spec!